quantum/process_keycode/process_autocorrect.c

```diff
@@ -1,36 +1,43 @@
 // Copyright 2021 Google LLC
 // Copyright 2021 @filterpaper
 // SPDX-License-Identifier: Apache-2.0
-// Original source: https://getreuer.info/posts/keyboards/autocorrection
+// Original source: https://getreuer.info/posts/keyboards/autocorrect
 
-#include "autocorrection.h"
+#include "process_autocorrect.h"
 #include <string.h>
+#include "keycode_config.h"
 
-#if __has_include("autocorrection_data.h")
-#    pragma GCC push_options
-#    pragma GCC optimize("O0")
-#    include "autocorrection_data.h"
-#    if AUTOCORRECTION_MIN_LENGTH < 4
-#        error Minimum Length is too short and may cause overflows
-#    endif
-#    if DICTIONARY_SIZE > SIZE_MAX
-#        error Dictionary size excees maximum size permitted
-#    endif
+#if __has_include("autocorrect_data.h")
+#    include "autocorrect_data.h"
+#else
+#    pragma message "Warning! Autocorrect is using the default library!"
+#    include "autocorrect_data_default.h"
+#endif
 
-bool process_autocorrection(uint16_t keycode, keyrecord_t* record) {
-    static uint8_t typo_buffer[AUTOCORRECTION_MAX_LENGTH] = {KC_SPC};
-    static uint8_t typo_buffer_size                       = 1;
+#if AUTOCORRECT_MIN_LENGTH < 5
+#    pragma message "Minimum Length may be too short and an cause incorrect behavior"
+#endif
+#if AUTOCORRECT_MIN_LENGTH < 4
+#    error Minimum Length is too short and may cause overflows
+#endif
+#if DICTIONARY_SIZE > SIZE_MAX
```

Comment:
It's defined by the toolchain and framework, actually.  LUFA and ChibiOS have it, and it a simple, brute force method to make sure it errors out when the database is too large.

Reply:
Specifically, `stdint.h` is what sets SIZE_MAX, and it the limit for the array size.  avr-gcc has a much lower value than arm-gcc.  So, what may work on ARM may not work on AVR (an issue I was actually running in (using a teensy 2++)

Reply:
To clarify, it will error out regardless if this check is in there or not.  But this adds some additional clarification as to why it's erroring out.  

If there is a better way to implement the check and message, i'm all for it

Reply:
Sounds good to me. Thanks for all those clarifications, this makes sense. SIZE_MAX is defined by stdint as the max possible value of size_t, which is also the largest allowed array size.

Regarding avr-gcc having a much lower SIZE_MAX, [this SO answer](https://stackoverflow.com/a/2550799/13223986) notes: "According to the 1999 ISO C standard (C99), size_t is an unsigned integer type of at least 16 bit (see sections 7.17 and 7.18.3)." So SIZE_MAX ought to be at least 64K, and this is a minor constraint on top of the existing "all nodes need to start within the first 64 KB" constraint.